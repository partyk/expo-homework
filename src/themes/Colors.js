export default {
    primary: '#474747',
    colorWhite: '#FFF',
    colorBlue: '#0badde',
    colorBlueLight: '#cdf3ff',
    colorBlueLighter: '#f5fcfe',
    colorPinkDark: '#cc517a',
    colorPink: '#f0bdd0',
    colorGreyDark: '#474747',
    colorGrey: '#bdc8cc',
    colorOrangeDark: '#ff390d',
    colorGreen: '#95c11f',
    colorBrownLight: '#c1b3b7',
}