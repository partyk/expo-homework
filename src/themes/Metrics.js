export default {
    buttons: {
        paddingHorizontal: 10,
        paddingVertical: 10,
        marginBottom: 10,
        height: 46,
    }
};
