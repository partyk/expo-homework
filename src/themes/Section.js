export default {
  primary: {
    paddingTop: 10,
    paddingHorizontal: 10
  }
}