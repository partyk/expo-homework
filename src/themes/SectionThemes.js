export default {
  primary: {
    paddingTop: 10,
    paddingHorizontal: 10
  },
  secondary: {
    paddingHorizontal: 10
  }
}