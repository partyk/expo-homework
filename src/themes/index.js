export Colors from './Colors';
export Metrics from './Metrics';
export Section from './Section';

// styleGuide
export ButtonThemes from './ButtonThemes';
