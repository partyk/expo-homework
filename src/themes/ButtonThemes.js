import Colors from './Colors';
import Metrics from './Metrics';

export default {
    primary: {
        button: {
            ...Metrics.buttons,
            alignItems: "center",
            justifyContent: "center",
            borderRadius: 5,
            backgroundColor: Colors.colorPinkDark,
        },
        text: {
            color: Colors.colorWhite,
            fontSize: 16
        }
    },
    secondary: {
        button: {
            ...Metrics.buttons,
            alignItems: "center",
            justifyContent: "center",
            borderRadius: 5,
            backgroundColor: Colors.colorBlue,
        },
        text: {
            color: Colors.colorWhite,
            fontSize: 16
        }
    }
};