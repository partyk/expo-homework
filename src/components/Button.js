// @flow
import React from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import ButtonThemes from '../themes/ButtonThemes';

const styles = StyleSheet.create({
    // default button is secondary
    button: ButtonThemes.secondary.button,
    text: ButtonThemes.secondary.text
});

type Props = {|
    +onPress: () => void,
    +children: string,
    +theme: string,
|}

export default class Button extends React.PureComponent<Props> {
    render() {
        const { children, onPress, theme } = this.props;
        return (
            <TouchableOpacity
                onPress={onPress}
                style={[styles.button, theme.button]}
            >
                <Text style={[styles.text, theme.text]}>{children}</Text>
            </TouchableOpacity>
        )
    }
}