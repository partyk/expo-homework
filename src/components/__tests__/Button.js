import React from 'react';
import renderer from 'react-test-renderer';

// components
import Button from '../Button';
import {ButtonThemes} from '../../themes';

test('Button primary renders correctly', () => {
    const tree = renderer
        .create(<Button onPress={() => null} theme={ButtonThemes.primary}>Button primary</Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

test('Button secondary renders correctly', () => {
    const tree = renderer
        .create(<Button onPress={() => null} theme={{}}>Button secondary</Button>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});