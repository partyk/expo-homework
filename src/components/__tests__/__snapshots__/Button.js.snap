// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button primary renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#cc517a",
      "borderRadius": 5,
      "height": 46,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
      ]
    }
  >
    Button primary
  </Text>
</View>
`;

exports[`Button secondary renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0badde",
      "borderRadius": 5,
      "height": 46,
      "justifyContent": "center",
      "marginBottom": 10,
      "opacity": 1,
      "paddingHorizontal": 10,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#FFF",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    Button secondary
  </Text>
</View>
`;
